### PR TITLE
test(scratchpad): run hashseed determinism checks concurrently

### PR DESCRIPTION
**Pull request by postmath**
#### What type of PR is this?

- [ ] bug
- [ ] feature
- [ ] documentation
- [x] cleanup

#### What this PR does:

Cuts `test_inplace_parent_choice_is_hashseed_independent` from ~57s to ~7s per
test method by running its subprocesses concurrently.

The test guards a real property — in-place parent selection must not depend on
set-iteration order — and it needs a fresh interpreter per seed, because
`PYTHONHASHSEED` only takes effect at startup. But it spawned 11 interpreters
*sequentially*: one baseline at seed 0, then seeds 1–9. Each one paid the full
cost of importing torch and the Inductor stack before it reached the solver,
because the snippet imports `torch_spyre._inductor`, whose `__init__` pulls in
`.patches` → `torch._inductor.graph` → dynamo → sympy. That import, not the
solver, was essentially the entire runtime.

The `-X importtime` breakdown, best-of-3 with a warm page cache:

| statement | wall |
|---|---|
| `pass` | 0.03s |
| `import torch` | 1.72s |
| `import torch_spyre` | 1.71s |
| `import torch_spyre._inductor` | **5.27s** |
| the snippet's actual imports | **5.51s** |

Note that `torch_spyre._inductor` costs more than torch itself does — the
`.patches` import adds ~3.6s on top. (`TORCH_DEVICE_BACKEND_AUTOLOAD=0`, which
the test already sets, does not help here: it disables torch auto-importing
torch_spyre via entry point, the opposite direction from what this snippet
does.)

Since the cost sits inside `subprocess.run`, threads release the GIL and the
seeds can all run at once. This PR:

- adds `_run_determinism_snippets()`, which fans the seeds out over a
  `ThreadPoolExecutor` and returns results in seed order;
- compares every result against the first, rather than running seed 0
  separately as a baseline;
- drops the seed count from 10 to 5;
- raises the per-subprocess timeout from 60s to 120s, since concurrent imports
  contend for I/O and CPU.

Wall time now collapses to roughly a single import. The check lives on the
`ScoreOrderingTests` mixin, which three solver suites inherit, so the file-level
saving is the one that matters:

| | before | after |
|---|---|---|
| one test method | 56.8s | ~6.8s |
| all three suites (`-k hashseed`) | ~170s | 20.3s |

#### Which issue(s) this PR is related to:

None — noticed while profiling the scratchpad solver test suite.

#### Special notes for your reviewer:

Reducing 10 seeds to 5 does trim coverage in principle. In practice set-iteration
order for the snippet's three buffer names changes under essentially any seed
perturbation, so a genuine ordering dependency surfaces within the first seed or
two; 10 bought little over 5.

I verified the assertion still bites rather than passing vacuously, by stubbing
`_run_determinism_snippet` to return a different placement for seed 3 — the test
fails and correctly reports `PYTHONHASHSEED=3`. Worth confirming, since the loop
now compares `results[0]` against itself on the first iteration.

One caveat on the timeout: on a cold page cache I measured a *single* import as
high as 43s wall (the venv here is on GPFS over NFS, so these imports are
I/O-bound — user time held steady at ~5.7s throughout). 120s held comfortably in
my runs, but if this test ever flakes in CI on a cold cache, that timeout is the
knob.

#### Does this PR introduce a user-facing change?

No. Test-only change; no library code touched.

#### Additional note:

`pre-commit run --files tests/inductor/test_scratchpad_solver.py` passes (ruff,
ruff format, mypy).

### CODE REVIEW

**Review by dgrove-oss — APPROVED** [review]
Summary

Speeds up test_inplace_parent_choice_is_hashseed_independent (~57s → ~7s per method) by fanning the per-PYTHONHASHSEED subprocess snippets out over a ThreadPoolExecutor instead of running them sequentially, since the dominant cost is import time inside each subprocess and threads release the GIL across subprocess.run. Also drops the seed count from 10 to 5 and raises the per-subprocess timeout from 60s to 120s to absorb contention from concurrent imports.

Review

BLOCKERs

- None found.

SUGGESTIONs

- None — the change is small, self-contained, and the author already verified the assertion still bites (stubbed a divergent result for one seed and confirmed the test fails with the correct PYTHONHASHSEED=N message).

QUESTIONs

- [ ] The PR description flags the 120s timeout as the one thing to watch if this ever flakes on a cold NFS/GPFS page cache — is there a plan to bump it further or add a retry if that happens, or is "come back and raise the timeout" the intended fallback? Not blocking, just want to confirm there's a known next step. (tests/inductor/test_scratchpad_solver.py:28)

Verdict

APPROVE

Test-only change (no library code touched), all CI checks pass, and DCO is satisfied (1/1 commit signed off). pool.map preserves input order regardless of completion order, so the zip(_DETERMINISM_HASHSEEDS, results) pairing is correct, and a subprocess failure (the assert p.returncode == 0 inside _run_determinism_snippet) correctly propagates out of pool.map rather than being silently dropped — verified both by reading the code and a quick local repro. The reduced seed count (10 → 5) is a reasonable coverage/cost tradeoff given the stated rationale (ordering violations, when present, surface within the first seed or two), and the timeout increase is justified by the now-concurrent I/O contention. No license, layout, dual-path, or work-division concerns apply.
